sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +246,855 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesIds, nameof(timeSeriesIds));
+
+                InstancesBatchRequest batchRequest = new InstancesBatchRequest()
+                {
+                    Get = new InstancesRequestBatchGetOrDelete()
+                };
+
+                foreach (TimeSeriesId timeSeriesId in timeSeriesIds)
+                {
+                    batchRequest.Get.TimeSeriesIds.Add(timeSeriesId);
+                }
+
+                Response<InstancesBatchResponse> executeBatchResponse = _timeSeriesInstancesRestClient
+                    .ExecuteBatch(batchRequest, _clientSessionId, cancellationToken);
+
+                return Response.FromValue(executeBatchResponse.Value.Get.ToArray(), executeBatchResponse.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Get search suggestion keywords based on Time Series instance attributes to be later used to search for instances asynchronously.
+        /// </summary>
+        /// <param name="searchString">The search string for which suggestions are required. Empty is allowed, but not null.</param>
+        /// <param name="maxNumberOfSuggestions">The maximum number of suggestions expected in the result. Defaults to 10 when not set.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A list of suggested search strings to be used for further search for Time Series instances or heirarchies.</returns>
+        /// <remarks>
+        /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/samples">our repo samples</see>.
+        /// </remarks>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleGetSearchSuggestions">
+        /// </code>
+        /// </example>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="searchString"/> is <c>null</c>.
+        /// </exception>
+        public virtual async Task<Response<SearchSuggestion[]>> GetSearchSuggestionsAsync(
+            string searchString,
+            int? maxNumberOfSuggestions = null,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetSearchSuggestions)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNull(searchString, nameof(searchString));
```

Comment:
Is it a service requirement that this can be empty but not null? I am wondering what we did in ADT for this kind of checks, add validation in sdk or let service throw an exception?

Reply:
You bring a good point. I looked at the Azure SDK design guidelines regarding parameter validations. Here's a useful link:
https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation

Given that searchString is passed directly to the service, it's considered a service parameter. So I will remove the assertion in this method and in the sync one.

Reply:
Thank you for the link, this is good to know.